### PR TITLE
TST: GH20676 Verify equals operator for list of Numpy arrays

### LINKED ISSUE #20676: pd.Series.equals() fails for Series containing iterable objects
#### Code Sample, a copy-pastable example if possible

```python
pd.Series(
    [np.array([1, 2]), np.array([1, 2])]
).equals(pd.Series(
    [np.array([1, 2]), np.array([1, 2])]
))

# Evaluates to False
```
#### Problem description

For objects that overload equality with elementwise equality (this includes `np.array` and `pd.Series` objects), `pd.Series.equals()` catches

```python
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

and returns `False`, behavior inherited from `np.equal` applied to object arrays. While you can argue this is a numpy bug, in numpy equality is purposefully strict for `dtype=='object'`. In pandas, it is a common enough procedure to tabulate arrays and lists under a single column (which may contain a named (time)series, image array, or an unpacked NLP document), and this behavior of equality is unintuitive.

#### Expected Output
`True`
#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 2.7.13.final.0
python-bits: 64
OS: Linux
OS-release: 4.4.0-1048-aws
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: None.None

pandas: 0.22.0
pytest: 3.2.2
pip: 9.0.1
setuptools: 27.2.0
Cython: None
numpy: 1.13.3
scipy: 0.19.1
pyarrow: None
xarray: None
IPython: 5.3.0
sphinx: 1.5.4
patsy: None
dateutil: 2.6.1
pytz: 2017.2
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 2.0.2
openpyxl: None
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: 4.6.0
html5lib: 0.9999999
sqlalchemy: 1.1.9
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by avinashpancham**
- [x] closes #20676
- [ x ] tests added / passed
- [ x ] passes `black pandas`
- [ x ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ x ] whatsnew entry

### CODE REVIEW

File: pandas/tests/series/methods/test_equals.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import numpy as np
+import pytest
+
+from pandas import Series
+
+
+@pytest.mark.parametrize(
+    "input_data", [[np.array([1, 2]), np.array([1, 2])]],
```
Comment (mroeschke) [inline pandas/tests/series/methods/test_equals.py:8]: You can in-line this data in the test since is parameterizing over 1 value

**Review by simonjayhawkins — APPROVED** [review]
Thanks @avinashpancham lgtm. as a follow-up test_equals in pandas\tests\generic\test_generic.py could be split and the Series tests moved here.

**Comment by avinashpancham** [comment]
@simonjayhawkins Moved the Series.equals tests from `generic/test_generic.py` to `series/methods/test_equals.py`. But I think I found a bug while restructuring the tests such that they are in the same style. This raises a ValueError instead of `False`:

```python
    arr = np.array([1, 2])
    s1 = Series([arr, arr])
    s2 = s1.copy()

    s1[1] = 9
   s1.equals(s2)
```

Expected:
False

Error
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

**Comment by jreback** [comment]
> @simonjayhawkins Moved the Series.equals tests from `generic/test_generic.py` to `series/methods/test_equals.py`. But I think I found a bug while restructuring the tests such that they are in the same style. This raises a ValueError instead of `False`. If this is a bug that should be addressed then I would like to help (if possible given my little experience).
> 
> **Code Sample, a copy-pastable example if possible**
> 
> ```python
> arr = np.array([1, 2])
> s1 = pd.Series([arr, arr])
> s2 = s1.copy()
> 
> s1[1] = 9
> s1.equals(s2)
> ```
> 
> **Problem description**
> Series.qquals operation fails when comparing two Series that have a single value and list at the same position.
> 
> ```
> ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
> ```
> 
> **Expected Output**
> False
> 
> ## **Output of pd.show_versions()**
> INSTALLED VERSIONS
> commit : [33df259](https://github.com/pandas-dev/pandas/commit/33df25994db637bc9a6988b4493b7db2410310a2)
> python : 3.8.3.final.0
> python-bits : 64
> OS : Darwin
> OS-release : 19.5.0
> Version : Darwin Kernel Version 19.5.0: Tue May 26 20:41:44 PDT 2020; root:xnu-6153.121.2~2/RELEASE_X86_64
> machine : x86_64
> processor : i386
> byteorder : little
> LC_ALL : None
> LANG : None
> LOCALE : None.UTF-8
> 
> Output of pd.show_versions()
> pandas : 1.1.0.dev0+2090.g33df25994.dirty
> numpy : 1.18.5
> pytz : 2020.1
> dateutil : 2.8.1
> pip : 20.1.1
> setuptools : 49.1.1.post20200710
> Cython : 0.29.21
> pytest : 5.4.3
> hypothesis : 5.19.0
> sphinx : 3.1.1
> blosc : None
> feather : None
> xlsxwriter : 1.2.9
> lxml.etree : 4.5.2
> html5lib : 1.1
> pymysql : None
> psycopg2 : None
> jinja2 : 2.11.2
> IPython : 7.16.1
> pandas_datareader: None
> bs4 : 4.9.1
> bottleneck : 1.3.2
> fsspec : 0.7.4
> fastparquet : 0.4.0
> gcsfs : 0.6.2
> matplotlib : 3.2.2
> numexpr : 2.7.1
> odfpy : None
> openpyxl : 3.0.4
> pandas_gbq : None
> pyarrow : 0.17.1
> pytables : None
> pyxlsb : None
> s3fs : 0.4.2
> scipy : 1.5.0
> sqlalchemy : 1.3.18
> tables : 3.6.1
> tabulate : 0.8.7
> xarray : 0.15.1
> xlrd : 1.2.0
> xlwt : 1.3.0
> numba : 0.50.1

@avinashpancham can you file a new issue about this on master. thanks.

**Comment by jreback** [comment]
thanks @avinashpancham

**Pull request merged**